mempool/reactor.go

```diff
@@ -60,12 +63,57 @@ func (memR *MempoolReactor) GetChannels() []*p2p.ChannelDescriptor {
 // AddPeer implements Reactor.
 // It starts a broadcast routine ensuring all txs are forwarded to the given peer.
 func (memR *MempoolReactor) AddPeer(peer p2p.Peer) {
+	// Getting the peer metric initializes it in the store
+	if memR.Switch != nil {
+		if tms := memR.Switch.MetricStore(); tms != nil {
+			_ = tms.GetPeerTrustMetric(peer.Key(), MempoolReactorID)
+		}
+	}
+
 	go memR.broadcastTxRoutine(peer)
 }
 
+// MarkPeer implements Reactor.
+// It updates a peer's metric with good or bad events taking place within this reactor
+func (memR *MempoolReactor) MarkPeer(peer p2p.Peer, good bool, events, severity int) {
```

Comment:
can these be deduplicated? seems like they're all the same except the reactor ID that gets passed at the end.

Also, what if we made Mark a method on Peer, rather than on Reactor? Is there some reason that won't work?

Reply:
I can move the mark method onto the Peer as long as the reactor is a parameter. This will require the ReactorIDs that are currently constants to be changed into fields within the reactor structs.

Reply:
Or, we could just pass the ReactorID to the peer mark method.

Reply:
Actually, we do need the reactor, since it has the reference to the Switch.

Reply:
Why does it need the switch, just for access to the metric store? Maybe there's a nice way to make that accessible within the peer ?

Reply:
Yes, for access to the metric store. Would you like the peer to have a metric store reference as well? In that case, we would just need to pass the reactor ID to the mark method.

Reply:
yeh I think that might make more sense - peers carry their own metric store

Reply:
great! it will be in the next commit